service/controller/v24/resource/workerasgname/resource.go

```diff
@@ -81,69 +79,25 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			cc.Status.Drainer.WorkerASGName = customObject.Status.AWS.AutoScalingGroup.Name
 
-			r.logger.LogCtx(ctx, "level", "debug", "message", "found the guest cluster worker ASG name in the CR")
-			return nil
-		}
-
-		r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster worker ASG name in the CR")
-	}
-
-	var workerASGName string
-	{
-		r.logger.LogCtx(ctx, "level", "debug", "message", "finding the guest cluster worker ASG name in the cloud formation stack")
-
-		stackOutputs, stackStatus, err := cc.CloudFormation.DescribeOutputsAndStatus(key.MainGuestStackName(customObject))
-		if cloudformationservice.IsStackNotFound(err) {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster worker ASG name in the cloud formation stack")
-			r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster main stack is not yet created")
-			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-			return nil
-
-		} else if cloudformationservice.IsOutputsNotAccessible(err) {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster worker ASG name in the cloud formation stack")
-			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("the guest cluster main stack output values are not accessible due to stack status '%s'", stackStatus))
+			r.logger.LogCtx(ctx, "level", "debug", "message", "found the tenant cluster worker ASG name in the CR")
 			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 			return nil
-
-		} else if err != nil {
-			return microerror.Mask(err)
-		}
-
-		workerASGName, err = cc.CloudFormation.GetOutputValue(stackOutputs, key.WorkerASGKey)
-		if cloudformationservice.IsOutputNotFound(err) {
-			// Since we are transitioning between versions we will have situations in
-			// which old clusters are updated to new versions and miss the ASG name in
-			// the CF stack outputs. We stop resource reconciliation at this point to
-			// reconcile again at a later point when the stack got upgraded and
-			// contains the ASG name.
-			//
-			// TODO remove this condition as soon as all guest clusters in existence
-			// obtain a ASG name.
-			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster worker ASG name in the cloud formation stack")
-			r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster main stack is not upgraded to the newest version yet")
-			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-			return nil
-
-		} else if err != nil {
-			return microerror.Mask(err)
 		}
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", "found the guest cluster worker ASG name in the cloud formation stack")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the tenant cluster worker ASG name in the CR")
 	}
 
-	// Store the ASG name in the CR status.
-	{
+	if cc.Status.Drainer.WorkerASGName != "" {
```

Comment:
I think we should have early return + log https://github.com/giantswarm/fmt/blob/master/go/log_messages.md#resource-intercommunication

Reply:
This here is the last block. There is only an return after it, nothing else. I tried putting the early return inside the block, but because after this block there is no code anymore the compiler complaint about unreachable code and then it was a bit weird.